pkg/sqlite/configmap_writer.go

```diff
@@ -0,0 +1,212 @@
+package sqlite
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+
+	"github.com/ghodss/yaml"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/operator-framework/operator-registry/pkg/client"
+)
+
+// configmap keys can contain underscores, but configmap names can not
+var unallowedKeyChars = regexp.MustCompile("[^-A-Za-z0-9_.]")
+
+const (
+	ResourceKey                 = "operators.coreos.com.bundle.resources"
+	MediaTypeKey                = "operators.coreos.com.bundle.mediatype"
+	EnvContainerImage           = "CONTAINER_IMAGE"
+	ConfigMapImageAnnotationKey = "olm.sourceImage"
+)
+
+type AnnotationsFile struct {
+	Annotations struct {
+		Resources string `json:"operators.coreos.com.bundle.resources"`
+		MediaType string `json:"operators.coreos.com.bundle.mediatype"`
+	} `json:"annotations"`
+}
+
+type ConfigMapWriter struct {
+	manifestsDir  string
+	configMapName string
+	namespace     string
+	clientset     *kubernetes.Clientset
+}
+
+var _ SQLPopulator = &ConfigMapWriter{}
+
+func NewConfigMapLoaderForDirectory(configMapName, namespace, manifestsDir, kubeconfig string) *ConfigMapWriter {
+	clientset, err := client.NewKubeClient(kubeconfig, logrus.StandardLogger())
+	if err != nil {
+		logrus.Fatalf("cluster config failed: %v", err)
+	}
+
+	return &ConfigMapWriter{
+		manifestsDir:  manifestsDir,
+		configMapName: configMapName,
+		namespace:     namespace,
+		clientset:     clientset,
+	}
+}
+
+func TranslateInvalidChars(input string) string {
+	validConfigMapKey := unallowedKeyChars.ReplaceAllString(input, "~")
+	return validConfigMapKey
+}
+
+func (c *ConfigMapWriter) Populate() error {
+	subDirs := []string{"/manifests/", "/metadata/"}
+
+	configMapPopulate, err := c.clientset.CoreV1().ConfigMaps(c.namespace).Get(c.configMapName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	configMapPopulate.Data = map[string]string{}
+
+	var totalSize uint64
+	for _, dir := range subDirs {
+		completePath := c.manifestsDir + dir
```

Comment:
Should you be using [path.Join](https://golang.org/pkg/path/#Join) here?

Reply:
Definitely.